### PR TITLE
build: enable in_kafka and out_kafka by default

### PR DESCRIPTION
**Pull request by tarruda**
close #5189

Signed-off-by: Thiago Padilha <thiago@calyptia.com>

### CODE REVIEW

**Comment by tarruda** [comment]
@agup006 I've enabled kafka plugins by default, but not on Windows since it causes a build error on CI

**Comment by patrick-stephens** [comment]
What's the build error?

**Comment by patrick-stephens** [comment]
> What's the build error? 

Looks like it doesn't like some of the void* usage: https://ci.appveyor.com/project/fluent/fluent-bit-2e87g/builds/43155542/job/yyye570bjrb67uvg#L2827

**Comment by tarruda** [comment]
Fixed the issue with windows build, just replacing `offsetof` by `container_of` macro took care of windows incompatibility

**Review by patrick-stephens — APPROVED** [review]
LGTM @niedbalski

**Comment by Boojapho** [comment]
@edsiper @fujimotos @koleini @leonardo-albertovich, could we get this reviewed/merged so the next release will have the kafka plugins enabled by default?

**Comment by patrick-stephens** [comment]
Currently there are merge conflicts, @tarruda can you resolve?

**Comment by patrick-stephens** [comment]
We also need docs to resolve #7269

**Comment by tarruda** [comment]
@patrick-stephens rebased.

**Comment by patrick-stephens** [comment]
@tarruda please disable for Windows if it does not compile: https://github.com/fluent/fluent-bit/blob/master/cmake/windows-setup.cmake

**Comment by tarruda** [comment]
@patrick-stephens rebased and fixed the builds. Couldn't enable by default on windows because the build fails on missing `libcrypto`, which I assume is a librdkafka dependency.

File: CMakeLists.txt [hunk 1]
```diff
@@ -236,7 +242,7 @@ option(FLB_OUT_NULL                    "Enable dev null output plugin"
 option(FLB_OUT_FLOWCOUNTER             "Enable flowcount output plugin"               Yes)
 option(FLB_OUT_LOGDNA                  "Enable LogDNA output plugin"                  Yes)
 option(FLB_OUT_LOKI                    "Enable Loki output plugin"                    Yes)
-option(FLB_OUT_KAFKA                   "Enable Kafka output plugin"                    No)
+option(FLB_OUT_KAFKA                   "Enable Kafka output plugin"                   ${FLB_KAFKA_ENABLE_DEFAULT})
```
Comment (leonardo-albertovich) [inline CMakeLists.txt:245]: This should be a plain `Yes`

File: CMakeLists.txt [hunk 2]
```diff
@@ -169,7 +175,7 @@ option(FLB_IN_FORWARD                  "Enable Forward input plugin"
 option(FLB_IN_HEALTH                   "Enable Health input plugin"                   Yes)
 option(FLB_IN_HTTP                     "Enable HTTP input plugin"                     Yes)
 option(FLB_IN_MEM                      "Enable Memory input plugin"                   Yes)
-option(FLB_IN_KAFKA                    "Enable Kafka input plugin"                     No)
+option(FLB_IN_KAFKA                    "Enable Kafka input plugin"                    ${FLB_KAFKA_ENABLE_DEFAULT})
```
Comment (leonardo-albertovich) [inline CMakeLists.txt:178]: This should be a plain `Yes`

File: CMakeLists.txt [hunk 3]
```diff
@@ -98,6 +98,12 @@ set(CMAKE_RUNTIME_OUTPUT_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}/bin")
 set(CMAKE_LIBRARY_OUTPUT_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}/lib")
 set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}/library")
 
+if(FLB_SYSTEM_WINDOWS)
```
Comment (leonardo-albertovich) [inline CMakeLists.txt:101]: The conditional disable should be done in the [windows specific cmake file](https://github.com/fluent/fluent-bit/blob/master/cmake/windows-setup.cmake#L6)

**Review by leonardo-albertovich — CHANGES_REQUESTED** [review]

**Comment by leonardo-albertovich** [comment]
These changes are a hard requirement for this PR to be merged.

**Review by leonardo-albertovich — APPROVED** [review]

**Comment by tarruda** [comment]
@leonardo-albertovich nevermind, it was a typo in windows-setup.cmake. Can you approve again?

**Review by leonardo-albertovich — APPROVED** [review]

**Pull request merged**